src/gossip/emitter.go

```diff
@@ -291,6 +291,11 @@ func (em *Emitter) createEvent() *inter.Event {
 			log.Crit("Emitter: head wasn't found", "e", p.String())
 		}
 		parentHeaders[i] = parent
+		if parentHeaders[i].Creator == coinbase && i != 0 {
```

Comment:
make sure the first parent is the self-parent. Ensure the condition as a post-condition of findBestParents.

Reply:
first parent is always self-parent. due to a fork, findBestParents could have found multiple self-parents